subsys/bluetooth/mesh/transport.c

```diff
@@ -1127,18 +1134,52 @@ static void seg_rx_reset(struct seg_rx *rx, bool full_reset)
 	}
 }
 
-static void seg_ack(struct k_work *work)
+static void seg_handler(struct k_work *work)
 {
-	struct seg_rx *rx = CONTAINER_OF(work, struct seg_rx, ack);
+	NET_BUF_SIMPLE_DEFINE(seg_buf, BT_MESH_RX_SDU_MAX);
```

Comment:
This will increase the stack size for the workqueue for all devices where the mesh message processing is the biggest consumer. Based on the workqueue stack sizes in the samples, this seems fairly unlikely, but it should definitely be part of the release notes, as it could cause overflow for existing apps with this 384 byte addition.

We previously moved this buffer off the stack to avoid this. It was on a different thread (on two threads even), but I'm a bit wary of putting something this big back on the stack.

Reply:
It does cause the stack space overhead of the system work queue to increase, but it does not mean that the 384-byte static allocation is completely transferred to the system stack. This will save some memory, but it does need to add a statement to let the developer know.